### ISSUE TITLE
can.Map define getter returning promise recomputing ad infinitum

### ISSUE THREAD

**Issue by m-mujica** [canjs/canjs#2248]
```js
modelPromise: {
  get: function() {
    var id = this.attr('id');
    
    return id === 'new' ?
      (new MyModel()).save() :
       MyModel.findOne({ id: id });
    }
}
```

that's how my getter looks like, if the value of `id` is `new` I can see the getter being recomputed each time the data returning from `Model.save()` is set to the model instance.

I have replicated the issue [here](http://jsbin.com/revaxapotu/1/edit?js,console)

**Comment by justinbmeyer** [comment]
Is this fixed by: https://github.com/canjs/canjs/issues/2220?

**Comment by m-mujica** [comment]
Fixed in 2.3.16 (I just tried it out and worked as expected).
